rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java

```diff
@@ -905,5 +907,67 @@ public Integer loadPreferredClearingDateLimit() {
         // returning default value 7 (days) if variable is not set
         return limit < 1 ? 7 : limit;
     }
+
+    public List<LicenseInfoParsingResult> processLicenseInfoWithObligations( List<LicenseInfoParsingResult> licenseInfoWithObligations, Map<String, String> releaseIdToAcceptedCLI, List<Release> releases, User user) throws TException {
+        ThriftClients thriftClients = new ThriftClients();
+        LicenseInfoService.Iface licenseClient = thriftClients.makeLicenseInfoClient();
+
+        for (Release release : releases) {
+            List<Attachment> approvedCliAttachments = SW360Utils.getApprovedClxAttachmentForRelease(release);
+            if (approvedCliAttachments.isEmpty()) {
+                approvedCliAttachments = SW360Utils.getClxAttachmentForRelease(release);
+            }
```

Comment:
Maybe put a notice in log that no approved CLX attached found. Going with attached CLX.

Reply:
comment addressed.